cluster-api/src/main/java/io/aiven/klaw/clusterapi/services/SchemaService.java

```diff
@@ -0,0 +1,208 @@
+        for (Integer schemaVersion : versionsList) {
+          String suffixUrl =
+              environmentVal + "/subjects/" + topicName + "-value/versions/" + schemaVersion;
+          Pair<String, RestTemplate> reqDetails =
+              clusterApiUtils.getRequestDetails(
+                  suffixUrl, protocol, KafkaClustersType.SCHEMA_REGISTRY);
+
+          Map<String, String> params = new HashMap<>();
+
+          ResponseEntity<HashMap> responseNew =
+              reqDetails.getRight().getForEntity(reqDetails.getLeft(), HashMap.class, params);
+          Map<String, Object> schemaResponse = responseNew.getBody();
+          if (schemaResponse != null) {
+            schemaResponse.put("compatibility", schemaCompatibility);
+          }
+
+          log.info(Objects.requireNonNull(responseNew.getBody()).toString());
+          allSchemaObjects.put(schemaVersion, schemaResponse);
+        }
+      }
+
+      return allSchemaObjects;
+    } catch (Exception e) {
+      log.error("Error from getSchema : " + e.getMessage());
+      return new TreeMap<>();
+    }
+  }
+
+  private List<Integer> getSchemaVersions(
+      String environmentVal, String topicName, KafkaSupportedProtocol protocol) {
+    try {
+      log.info("Into getSchema versions {} {}", topicName, environmentVal);
+      if (environmentVal == null) {
+        return null;
+      }
+
+      String suffixUrl = environmentVal + "/subjects/" + topicName + "-value/versions";
+      Pair<String, RestTemplate> reqDetails =
+          clusterApiUtils.getRequestDetails(suffixUrl, protocol, KafkaClustersType.SCHEMA_REGISTRY);
+
+      Map<String, String> params = new HashMap<>();
+
+      ResponseEntity<ArrayList> responseList =
+          reqDetails.getRight().getForEntity(reqDetails.getLeft(), ArrayList.class, params);
+      log.info("Schema versions " + responseList);
+      return responseList.getBody();
+    } catch (Exception e) {
+      log.error("Error in getting versions " + e.getMessage());
+      return new ArrayList<>();
+    }
+  }
+
+  private String getSchemaCompatibility(
+      String environmentVal, String topicName, KafkaSupportedProtocol protocol) {
+    try {
+      log.info("Into getSchema compatibility {} {}", topicName, environmentVal);
+      if (environmentVal == null) {
+        return null;
+      }
+
+      String suffixUrl = environmentVal + "/config/" + topicName + "-value";
+      Pair<String, RestTemplate> reqDetails =
+          clusterApiUtils.getRequestDetails(suffixUrl, protocol, KafkaClustersType.SCHEMA_REGISTRY);
+
+      Map<String, String> params = new HashMap<>();
+
+      ResponseEntity<HashMap> responseList =
+          reqDetails.getRight().getForEntity(reqDetails.getLeft(), HashMap.class, params);
+      log.info("Schema compatibility " + responseList);
+      return (String) responseList.getBody().get("compatibilityLevel");
+    } catch (Exception e) {
+      log.error("Error in getting schema compatibility " + e.getMessage());
```

Comment:
```suggestion
      log.error("Error in getting schema compatibility ", e);
```
it's better to log the whole trace